### PR TITLE
build the rocksdb library portably

### PR DESCRIPTION
**Pull request by marpaia**
Adding an environment variable as illustrated in the compilation section of the install docs: https://github.com/facebook/rocksdb/blob/master/INSTALL.md

### CODE REVIEW

**Comment by marpaia** [comment]
CC @igorcanadi

File: Library/Formula/rocksdb.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Rocksdb < Formula
 
   def install
     ENV.cxx11
+    ENV["PORTABLE"] = 1
```
Comment (tdsmith) [inline Library/Formula/rocksdb.rb]: Let's make this conditional on `build.bottle?`.

Comment (marpaia) [inline Library/Formula/rocksdb.rb]: like this?

```rb
if build.bottle? do
  ENV["PORTABLE"] = 1
end
```

Comment (DomT4) [inline Library/Formula/rocksdb.rb]: Don't need to wrap it. `ENV["PORTABLE"] = 1 if build.bottle?` should be fine.

Comment (marpaia) [inline Library/Formula/rocksdb.rb]: updated the PR. you ruby folks and your one liners :)

**Comment by tdsmith** [comment]
Just to make sure the effect of this is what you want: if you install the Homebrew-distributed bottle or install from source with --build-bottle, you'll get the portable build. If you build from source without specifying --build-bottle, this will give you a -march=native build.

File: Library/Formula/rocksdb.rb (hunk 1 above)
Comment (tdsmith) [inline Library/Formula/rocksdb.rb]: We're irrepressible!

File: Library/Formula/rocksdb.rb [hunk 2]
```diff
@@ -18,6 +18,7 @@ class Rocksdb < Formula
 
   def install
     ENV.cxx11
+    ENV["PORTABLE"] = 1 if build.bottle?
```
Comment (tdsmith) [inline Library/Formula/rocksdb.rb]: I think this needs to be "1"

**Comment by marpaia** [comment]
@tdsmith correct, except for it's the `PORTABLE=1` flag which adds the `-march=native` compiler option: https://github.com/facebook/rocksdb/blob/98a44559d590a8cb63551393e275b57ed7e54504/build_tools/build_detect_platform#L334-L336

I'm running this by someone on the RocksDB team right now just to make sure everything is OK with them, I'll report back shortly.

**Comment by marpaia** [comment]
@tdsmith got the thumbs up from the team. feel free to merge whenever you're ready.

**Comment by siying** [comment]
I'm not expert on this and neither does most people in RocksDb team. We can merge whatever you guys think appropriate.

**Comment by tdsmith** [comment]
`test -z "$PORTABLE"` tests whether `"$PORTABLE"` is zero-length; build_detect_platform will add `-march=native` when `$PORTABLE` is not set (which matches my expectations, since native-optimized binaries are non-portable).

**Comment by igorcanadi** [comment]
> test -z "$PORTABLE" tests whether "$PORTABLE" is zero-length; build_detect_platform will add -march=native when $PORTABLE is not set (which matches my expectations, since native-optimized binaries are non-portable).

This is true. We add -march=native when $PORTABLE is not set.

**Comment by tdsmith** [comment]
Pulled in dd629a; thanks for the PR!

**Pull request closed without merging**